benchmarking/plugins/MarlinBenchHistNativProcessor.cc

```diff
@@ -0,0 +1,71 @@
+#include <vector>
+#include <list>
+#include <random>
+
+// -- marlin headers
+#include <marlin/Processor.h>
+#include <marlin/ProcessorApi.h>
+#include <marlin/PluginManager.h>
+
+// -- root headers
+#include <ROOT/RHist.hxx>
+#include <ROOT/RHistConcurrentFill.hxx>
+
+using namespace marlin ;
+using namespace ROOT::Experimental ;
+
+namespace marlin{
+  using size_t = unsigned long long int;
+}
+
+class MarlinBenchHistNativeProcessor : public Processor {
+public:
+  MarlinBenchHistNativeProcessor() ;
+  void init() final override;
+  void processEvent( EventStore * evt) final override;
+  void end() final override;
+
+private:
+  Property<int> _nBins {this, "NBins",  "number of bins per histogram.", 1000} ;
+  Property<int> _nHist {this, "NHists", "number of histograms to create.", 5} ;
+  Property<marlin::size_t> _nFills {this, "NFills", "number of fill operations per event", 1000} ;
+  std::vector<RH1F> _histograms ;
+  std::list<RHistConcurrentFillManager<RH1F, 1024>> _managers;  
+};
+
+MarlinBenchHistNativeProcessor::MarlinBenchHistNativeProcessor() :
+  Processor("MarlinBenchHistNativeProcessor") {}
+
+void MarlinBenchHistNativeProcessor::init() {
+  if (_histograms.empty()) {
+    for (int i = 0; i < _nHist; ++i) {
+      _histograms.emplace_back(
+          "dummy histogram", 
+          RAxisConfig(  
+            "normal distributed values", 
+            static_cast<int>(_nBins), 
+            -20.f, 20.f));
+      _managers.emplace_back(_histograms.back());
+    }
+  }
+}
+
+void MarlinBenchHistNativeProcessor::processEvent(EventStore * evt) {
+  auto seed = ProcessorApi::getRandomSeed( this, evt ) ;
```

Comment:
If you get a random seed, you need to register the processor to get a random seed using the correct function from `ProcessorApi`. Call it in the `init` function.